.github/workflows/protocol_tests.yml

```diff
@@ -94,11 +112,32 @@ jobs:
             exit 1
           fi
 
-      - name: Run everything just in case something was missed (excl. integration and e2e tests)
+      - name: Run all unit tests in case some were missed (excl. integration and e2e tests)
         # can't use gas limit because some setUp function use more than the limit
-        # Excludes integration tests, because they require an anvil devchain running on the localhost.
-        run: forge test -vvv --no-match-contract "RegistryIntegrationTest|E2EDemo"
+        # Excludes e2e and integration tests, because they require a connection to an anvil devchain
+        # serving at localhost.
+        run: |
+          forge test -vvv \
+          --match-path "test-sol/unit/*"
+      
+      - name: Generate migrations and run devchain
+        if: success() || failure()
+        run: ./scripts/foundry/create_and_migrate_anvil_devchain.sh
+      
+      - name: Run integration tests against local anvil devchain
+        if: success() || failure()
+        run: |
+          forge test -vvv \
+          --match-path "test-sol/integration/*" \
+          --fork-url http://127.0.0.1:${{ env.ANVIL_PORT }}
       
-      - name: Generate migrations
+      - name: Run e2e tests against local anvil devchain
         if: success() || failure()
-        run: ./migrations_sol/run_integration_tests_in_anvil.sh
+        run: |
+          forge test -vvv \
+          --match-path "test-sol/e2e/*" \
+          --fork-url http://127.0.0.1:${{ env.ANVIL_PORT }}
+
+      # TODO(Arthur): It does not seem necessary to kill Anvil since the GitHub workflow 
```

Comment:
Is this TODO still needed?

Not necessary for running CI tests, but would be good to have when running locally. otherwise, the devchain stays running in the background. 

might be worth creating a wrapper bash script around `./scripts/foundry/create_and_migrate_anvil_devchain.sh` that keeps the devchain running in the foreground, it can easily be close.

A yarn cmd to run the foundry devchain would also be nice.

Reply:
> Is this TODO still needed?

Great catch, thank you. The comment is not needed anymore. I removed it in https://github.com/celo-org/celo-monorepo/pull/11023/commits/61b6d7d64e09351635f5d6da509008d132e28a99.

Reply:
> Not necessary for running CI tests, but would be good to have when running locally. otherwise, the devchain stays running in the background.
> 
> might be worth creating a wrapper bash script around ./scripts/foundry/create_and_migrate_anvil_devchain.sh that keeps the devchain running in the foreground, it can easily be close.
> 
> A yarn cmd to run the foundry devchain would also be nice.

Great point, I added two yarn commands to start and stop the devchain when working locally:

1. `yarn anvil-devchain:start` starts a new anvil devchain serving at localhost.

    For example:
    
    ```sh
    $ yarn anvil-devchain:start
    
    yarn run v1.22.22
    $ ./scripts/foundry/create_and_migrate_anvil_devchain.sh
    # ...
    Total elapsed time: 193 seconds
    ✨  Done in 193.09s.
    ```

1. `yarn anvil-devchain:stop` terminates any anvil nodes serving at localhost.


    For example:

    ```sh
    # in packages/protocol/ directory
    $ yarn anvil-devchain:stop
    
    yarn run v1.22.22
    $ ./scripts/foundry/stop_anvil.sh
    Connection to localhost port 8546 [tcp/*] succeeded!
    Killed Anvil
    ✨  Done in 0.11s.
    ```

Reply:
Also added a `CONTRIBUTING.md` file in https://github.com/celo-org/celo-monorepo/pull/11023/commits/bd61a72cf8ee31d406a71985696ff40aa33d2da4